From 56b1a3dc6ec0cf59a487d732d2fa0b805398b454 Mon Sep 17 00:00:00 2001
From: "hayim@marvell.com" <hayim@marvell.com>
Date: Mon, 29 Feb 2016 16:22:26 +0200
Subject: [PATCH 0907/1240] a8k: remove doimage during build process

doimage part was moved to ATF. Therefore, it was
removed from A8K u-boot build process

Change-Id: Ic00ea882c9b3cb0f6107176ae712292fd1355228
Signed-off-by: hayim@marvell.com <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27936
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 Makefile | 55 +++++--------------------------------------------------
 1 file changed, 5 insertions(+), 50 deletions(-)

diff --git a/Makefile b/Makefile
index 8b92489..d4affc5 100644
--- a/Makefile
+++ b/Makefile
@@ -732,9 +732,6 @@ ALL-$(CONFIG_REMAKE_ELF) += u-boot.elf
 # We can't do this yet due to the need for binary blobs
 # ALL-$(CONFIG_X86_RESET_VECTOR) += u-boot.rom
 
-ALL-$(CONFIG_MVEBU) += doimage
-ALL-$(CONFIG_PALLADIUM) += bin2phex
-
 # enable combined SPL/u-boot/dtb rules for tegra
 ifneq ($(CONFIG_TEGRA),)
 ifeq ($(CONFIG_SPL),y)
@@ -840,53 +837,7 @@ u-boot.ldr.hex u-boot.ldr.srec: u-boot.ldr FORCE
 ifdef CONFIG_MVEBU
 SPLIMAGE	:= $(srctree)/spl/u-boot-spl.bin
 
-ifeq ($(CONFIG_TARGET_ARMADA_8K), y)
-
-DOIMAGE		:=  $(srctree)/tools/doimage
-BIN2PHEX	:= $(srctree)/scripts/bin2phex.pl
-DOIMAGE_SEC	:= $(srctree)/tools/secure/sec_img.cfg
-
-ifdef CONFIG_MVEBU_SECURE_BOOT
-
-DOIMAGE_SEC_FLAGS := -c $(DOIMAGE_SEC)
-DOIMAGE_LIBS_CHECK = \
-	if ! [ -d "/usr/include/polarssl" ]; then \
-			echo "****************************************" >&2; \
-			echo "Missing PolarSSL installation! " >&2; \
-			echo "Please download it from \"tls.mbed.org\"" >&2; \
-			echo "Make sure to use version 1.3.11 or later" >&2; \
-			echo "****************************************" >&2; \
-		exit 1; \
-	else if ! [ -f "/usr/include/libconfig.h" ]; then \
-			echo "********************************************************" >&2; \
-			echo "Missing Libconfig installation!" >&2; \
-			echo "Please download it from \"www.hyperrealm.com/libconfig/\"" >&2; \
-			echo "Alternatively on Debian/Ubuntu system install packages" >&2; \
-			echo "\"libconfig8\" and \"libconfig8-dev\"" >&2; \
-			echo "********************************************************" >&2; \
-		exit 1; \
-	fi \
-	fi
-else
-DOIMAGE_LIBS_CHECK =
-DOIMAGE_SEC_FLAGS =
-
-endif #CONFIG_MVEBU_SECURE_BOOT
-
-ifdef CONFIG_MVEBU_NAND_BLOCK_SIZE
-NAND_DOIMAGE_FLAGS := -t $(CONFIG_MVEBU_NAND_CELL_TYPE) -n $(CONFIG_MVEBU_NAND_BLOCK_SIZE)
-endif
-DOIMAGE_FLAGS	:= -b $(SPLIMAGE) $(NAND_DOIMAGE_FLAGS) $(DOIMAGE_SEC_FLAGS)
-
-doimage: $(obj)/u-boot.bin $(DOIMAGE) $(SPLIMAGE)
-		@$(DOIMAGE_LIBS_CHECK)
-		$(DOIMAGE) $(DOIMAGE_FLAGS) u-boot.bin u-boot-spl.bin
-
-bin2phex: doimage
-		$(BIN2PHEX) -w 1  -i u-boot-spl.bin -o u-boot-spl.hex -b 0x0
-		$(BIN2PHEX) -w 16 -i u-boot.bin -o u-boot.hex -b 0x0
-
-else ifeq ($(CONFIG_TARGET_ARMADA_3700), y)
+ifeq ($(CONFIG_TARGET_ARMADA_3700), y)
 
 BIN2PHEX	:= $(srctree)/scripts/bin2phex.pl
 TIM2PHEX	:= $(srctree)/scripts/tim2phex.pl
@@ -941,6 +892,8 @@ DOIMAGE_LIBS_CHECK = \
 		echo "DOIMAGE=$(DOIMAGE)" >&1; \
 	fi
 
+ALL-$(CONFIG_MVEBU) += doimage
+
 # Start with creation of UART images:
 # - Create TIM descriptor with UART signature
 # - Create binary TIM and UART downloadable images (*_h.*)
@@ -1005,6 +958,8 @@ MKIMAGEFLAGS_u-boot.kwb = -n $(srctree)/$(CONFIG_SYS_KWD_CONFIG:"%"=%) \
 MKIMAGEFLAGS_u-boot.pbl = -n $(srctree)/$(CONFIG_SYS_FSL_PBL_RCW:"%"=%) \
 		-R $(srctree)/$(CONFIG_SYS_FSL_PBL_PBI:"%"=%) -T pblimage
 
+ALL-$(CONFIG_MVEBU) += doimage
+
 u-boot.img u-boot.kwb u-boot.pbl: u-boot.bin FORCE
 	$(call if_changed,mkimage)
 
-- 
1.9.1

